pkg/globalnet/controllers/packetfilter/iface.go

```diff
@@ -0,0 +1,369 @@
+var logger = log.Logger{Logger: logf.Log.WithName("PacketFilter")}
+
+func New() (Interface, error) {
+	pFilterHandler, err := packetfilter.New()
+	if err != nil {
+		return nil, err //nolint:wrapcheck  // Let the caller wrap it
+	}
+
+	pGNEgressHandler, err := packetfilter.NewGlobalNetEgress()
+	if err != nil {
+		return nil, err //nolint:wrapcheck  // Let the caller wrap it
+	}
+
+	pFilterIface := &pfilter{
+		pFilter:   pFilterHandler,
+		pGNEgress: pGNEgressHandler,
+	}
+
+	return pFilterIface, nil
+}
+
+func (i *pfilter) AddClusterEgressRules(subnet, snatIP, globalNetIPTableMark string) error {
+	ruleSpec := packetfilter.Rule{
+		Proto:     packetfilter.RuleProtoAll,
+		SrcCIDR:   subnet,
+		MarkValue: globalNetIPTableMark,
+		SnatCIDR:  snatIP,
+		Action:    packetfilter.RuleActionSNAT,
+	}
+
+	if err := i.pFilter.AppendUnique(packetfilter.TableTypeNAT, constants.SmGlobalnetEgressChainForCluster, &ruleSpec); err != nil {
+		return errors.Wrapf(err, "unable to append rule %+v", &ruleSpec)
+	}
+
+	return nil
+}
+
+func (i *pfilter) RemoveClusterEgressRules(subnet, snatIP, globalNetIPTableMark string) error {
+	ruleSpec := packetfilter.Rule{
+		Proto:     packetfilter.RuleProtoAll,
+		SrcCIDR:   subnet,
+		MarkValue: globalNetIPTableMark,
+		SnatCIDR:  snatIP,
+		Action:    packetfilter.RuleActionSNAT,
+	}
+
+	if err := i.pFilter.Delete(packetfilter.TableTypeNAT, constants.SmGlobalnetEgressChainForCluster, &ruleSpec); err != nil {
+		return errors.Wrapf(err, "error deleting packetfilter rule %+v", &ruleSpec)
+	}
+
+	return nil
+}
+
+func (i *pfilter) AddIngressRulesForHeadlessSvc(globalIP, ip string, targetType TargetType) error {
+	if globalIP == "" || ip == "" {
+		return fmt.Errorf("globalIP %q or %s IP %q cannot be empty", globalIP, targetType, ip)
+	}
+
+	ruleSpec := packetfilter.Rule{
+		DestCIDR: globalIP,
+		DnatCIDR: ip,
+		Action:   packetfilter.RuleActionDNAT,
+	}
+	logger.V(log.DEBUG).Infof("Installing packetfilter rule for Headless SVC %+v for %s", &ruleSpec, targetType)
+
+	if err := i.pFilter.AppendUnique(packetfilter.TableTypeNAT, constants.SmGlobalnetIngressChain, &ruleSpec); err != nil {
+		return errors.Wrapf(err, "unable to append rule %+v", &ruleSpec)
+	}
+
+	return nil
+}
+
+func (i *pfilter) RemoveIngressRulesForHeadlessSvc(globalIP, ip string, targetType TargetType) error {
+	if globalIP == "" || ip == "" {
+		return fmt.Errorf("globalIP %q or %s IP %q cannot be empty", globalIP, targetType, ip)
+	}
+
+	ruleSpec := packetfilter.Rule{
+		DestCIDR: globalIP,
+		DnatCIDR: ip,
+		Action:   packetfilter.RuleActionDNAT,
+	}
+	logger.V(log.DEBUG).Infof("Deleting iptables rule for Headless SVC %+v for %s", &ruleSpec, targetType)
+
+	if err := i.pFilter.Delete(packetfilter.TableTypeNAT, constants.SmGlobalnetIngressChain, &ruleSpec); err != nil {
+		return errors.Wrapf(err, "error deleting packetfilter rule %+v", &ruleSpec)
+	}
+
+	return nil
+}
+
+func (i *pfilter) GetKubeProxyClusterIPServiceChainName(service *corev1.Service,
```

Comment:
We no longer depend on the kubeproxy serviceChain for supporting GN. This API can be deleted.